File: backend/app/schemas/review.py
# ============================================================
# 复盘 schema（对齐前端 types/review.ts）
# ============================================================
from __future__ import annotations

from datetime import date, datetime

from pydantic import BaseModel, Field


class ReviewReflection(BaseModel):
    question: str
    answer: str = ""


class ReviewData(BaseModel):
    completed_tasks: str = ""
    unfinished_tasks: str = ""
    gains: str = ""
    reflections: list[ReviewReflection] = []
    tomorrow_plan: str = ""
    mood: int = Field(default=3, ge=1, le=5)
    energy: int = Field(default=3, ge=1, le=5)


class ReviewOut(BaseModel):
    id: str
    type: str
    review_date: date
    project_id: str | None = None
    data: ReviewData
    created_at: datetime
    updated_at: datetime


class UpsertReviewRequest(BaseModel):
    type: str = "daily"
    date: date
    project_id: str | None = None
    data: ReviewData


class ReviewListItem(BaseModel):
    id: str
    type: str
    review_date: date
    mood: int | None = None
    energy: int | None = None
    summary: str = ""
    created_at: datetime
    updated_at: datetime


class AutoFillTask(BaseModel):
    id: str
    title: str
    completed_at: datetime | None = None
    due_date: date | None = None


class AutoFillDoc(BaseModel):
    id: str
    title: str
    created_at: datetime


class AutoFillStats(BaseModel):
    tasks_completed: int = 0
    documents_created: int = 0
    tasks_overdue: int = 0


class AutoFillData(BaseModel):
    completed_tasks: list[AutoFillTask] = []
    unfinished_tasks: list[AutoFillTask] = []
    documents_created: list[AutoFillDoc] = []
    stats: AutoFillStats = Field(default_factory=AutoFillStats)


class ConvertTaskRequest(BaseModel):
    content: str = Field(min_length=1)
    priority: str | None = None
    due_date: date | None = None

File: frontend/src/types/review.ts
// ============================================================
// 复盘模块类型
// 对应 PRD §13.6 复盘模块 + §15.10 reviews表
// ============================================================

export type ReviewType = 'daily' | 'weekly' | 'monthly'

export interface ReviewData {
  completed_tasks: string
  unfinished_tasks: string
  gains: string
  reflections: Array<{ question: string; answer: string }>
  tomorrow_plan: string
  mood: number
  energy: number
}

export interface Review {
  id: string
  type: ReviewType
  review_date: string
  data: ReviewData
  created_at: string
  updated_at: string
}

export interface ReviewListQuery {
  type?: ReviewType
  page?: number
  page_size?: number
}

export interface UpsertReviewRequest {
  type: ReviewType
  date: string
  data: ReviewData
}

export interface AutoFillData {
  completed_tasks: Array<{ id: string; title: string; completed_at: string }>
  unfinished_tasks: Array<{ id: string; title: string; due_date: string | null }>
  documents_created: Array<{ id: string; title: string; created_at: string }>
  stats: {
    tasks_completed: number
    documents_created: number
    tasks_overdue: number
  }
}

export interface ConvertTaskRequest {
  content: string
  priority?: string
  due_date?: string
}
